dozer-sql/expression/src/scalar/string.rs

```diff
@@ -272,6 +272,200 @@ pub(crate) fn evaluate_to_char(
     Ok(Field::String(output))
 }
 
+pub(crate) fn evaluate_chr(
+    schema: &Schema,
+    arg: &mut Expression,
+    record: &Record,
+) -> Result<Field, Error> {
+    let value = arg.evaluate(record, schema)?;
+    match value {
+        Field::UInt(u) => Ok(Field::String((((u % 256) as u8) as char).to_string())),
+        Field::U128(u) => Ok(Field::String((((u % 256) as u8) as char).to_string())),
+        Field::Int(i) => Ok(Field::String((((i % 256) as u8) as char).to_string())),
```

Comment:
The behavior for negative arguments is not well-defined. In oracle, a negative argument results in a `numeric overflow` error. Let's error on negatives.

Furthermore, this assumes ascii. We don't support non-utf8 character sets within dozer. For arguments in the range [0, 255], the correct behavior is obvious. Outside that, in rust, valid values would be  [0, 0x10FFFF], excluding [0xD800, 0xDFFF].

Let's keep the `mod 256` behavior, but error on negative values and warn on values larger than 256?

Reply:
Agree on negatives, the documentation was not clear on that.
What you mean by "larger than 256"?

Reply:
The oracle documentation mentions that for single-byte encodings, the effective value will be `mod 256`. For multi-byte encodings, like unicode encodings, that's not the case. For example, for UTF-8, like rust's strings, valid values are [0, 0x10FFFF], excluding [0xD800, 0xDFFF].